recipes/quickstart/agents/Agents_101/Tool_Calling_101.ipynb

```diff
@@ -0,0 +1,980 @@
+    "                \"default\": \"none\"\n",
+    "                }\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "]\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "system_prompt = \"\"\"You are an expert in composing functions. You are given a question and a set of possible functions. \n",
+    "Based on the question, you will need to make one or more function/tool calls to achieve the purpose. \n",
+    "If none of the function can be used, point it out. If the given question lacks the parameters required by the function,\n",
+    "also point it out. You should only return the function call in tools call sections.\n",
+    "\n",
+    "If you decide to invoke any of the function(s), you MUST put it in the format of [func_name1(params_name1=params_value1, params_name2=params_value2...), func_name2(params)]\\n\n",
+    "You SHOULD NOT include any other text in the response.\n",
+    "\n",
+    "Here is a list of functions in JSON format that you can invoke.\\n\\n{functions}\\n\"\"\".format(functions=function_definitions)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chat_history = []\n",
+    "\n",
+    "def model_chat(user_input: str, sys_prompt = system_prompt, temperature: int = 0.7, max_tokens=2048):\n",
+    "    \n",
+    "    chat_history = [\n",
+    "        {\n",
+    "            \"role\": \"system\",\n",
+    "            \"content\": system_prompt\n",
+    "        }\n",
+    "    ]\n",
+    "    \n",
+    "    chat_history.append({\"role\": \"user\", \"content\": user_input})\n",
+    "    \n",
+    "    #print(chat_history)\n",
+    "    \n",
+    "    #print(\"User: \", user_input)\n",
+    "    \n",
+    "    response = client.chat.completions.create(model=\"llama-3.2-3b-preview\",\n",
+    "                                          messages=chat_history,\n",
+    "                                          max_tokens=max_tokens,\n",
+    "                                          temperature=temperature)\n",
+    "    \n",
+    "    chat_history.append({\n",
+    "    \"role\": \"assistant\",\n",
+    "    \"content\": response.choices[0].message.content\n",
+    "    })\n",
+    "    \n",
+    "    \n",
+    "    #print(\"Assistant:\", response.choices[0].message.content)\n",
+    "    \n",
+    "    return response.choices[0].message.content"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Assistant: [get_user_info(user_id=7890, special='black')]\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_input = \"Can you retrieve the details for the user with the ID 7890, who has black as their special request?\"\n",
```

Comment:
`who has black as their special request` can you pls elaborate a bit about this prompt. More specifically what does black mean in this prompt.